### PR TITLE
KOGITO-2427: Milestones panel on process details page

### PR DESCRIPTION
**Pull request by Sara4994**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/ProcessDetailsMilestones.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.kogito-management-console--milestones__nameText > blockquote{
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/ProcessDetailsMilestones.css]: not sure - maybe whitespace format? before '{'

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/tests/__snapshots__/ProcessDetailsMilestones.test.tsx.snap [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+exports[`Process details page milestones panel Snapshot test with default props 1`] = `
+<ProcessDetailsMilestones
+  milestones={
+    Array [
+      Object {
+        "__typename": "Milestones",
+        "id": "27107f38-d888-4edf-9a4f-11b9e6d75m36",
+        "name": "Milestone 1: Order placed",
+        "status": undefined,
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/tests/__snapshots__/ProcessDetailsMilestones.test.tsx.snap]: statuses here are undefined, thus there are no labels for the milestones below.

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/ProcessDetailsMilestones.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  milestones?: Pick<Milestone, 'id' | 'name' | 'status'>[];
+}
+const ProcessDetailsMilestones: React.FC<IOwnProps & OUIAProps> = ({
+  ouiaId,
+  ouiaSafe,
+  milestones
+}) => {
+  const handleStatus = status => {
+    switch (status) {
+      case 'AVAILABLE':
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/ProcessDetailsMilestones.tsx:28]: in graphql types.tsx I can see just available, active, completed. Terminated is not there.

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/ProcessDetailsMilestones.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+          <Label color="orange" icon={<InfoCircleIcon />}>
+            Terminated
+          </Label>
+        );
+      default:
+        break;
+    }
+  };
+  const compareObjs = (firstEle, secondEle) => {
+    if (firstEle.status < secondEle.status) {
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/ProcessDetailsMilestones.tsx:47]: I guess the comparison is based on the order in which values are defined in MilestoneStatus type?

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/tests/ProcessDetailsMilestones.test.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+import { GraphQL } from '@kogito-apps/common';
+import MilestoneStatus = GraphQL.MilestoneStatus;
+// tslint:disable: no-string-literal
+describe('Process details page milestones panel', () => {
+  const props = {
+    milestones: [
+      {
+        id: '27107f38-d888-4edf-9a4f-11b9e6d75m36',
+        name: 'Milestone 1: Order placed',
+        status: MilestoneStatus['ACTIVE'],
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/tests/ProcessDetailsMilestones.test.tsx]: the status values are probably not passed correctly, see the snapshot, it results in undefined, thus there's no value returned by handleStatus function.

File: ui-packages/packages/management-console/src/components/Templates/ProcessDetailsPage/tests/__snapshots__/ProcessDetailsPage.test.tsx.snap [hunk 6]
```diff
@@ -1041,1886 +1152,8 @@ exports[`Process Details Page component tests snapshot testing in Active state 1
...
-              </Portal>
-            </Modal>
-          </div>
-        </Flex>
-      </section>
-    </PageSection>
-  </div>
-</ProcessDetailsPage>
-`;
+exports[`Process Details Page component tests snapshot testing in Error state 1`] = `null`;
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Templates/ProcessDetailsPage/tests/__snapshots__/ProcessDetailsPage.test.tsx.snap]: broken tests.

**Review by jstastny-cz — CHANGES_REQUESTED** [review]

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/ProcessDetailsMilestones.tsx (hunk 3 above)
Comment (Sara4994) [reply ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/ProcessDetailsMilestones.tsx:28]: Terminated will be added in types.tsx. @nmirasch wdyt?

Comment (nmirasch) [reply ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/ProcessDetailsMilestones.tsx:28]: @Sara4994, @jstastny-cz  yes that was my fault on the jira description I have added that terminated status. But now is confirmed https://issues.redhat.com/browse/KOGITO-2427?focusedCommentId=14417836&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14417836 we have available, active, completed status.

File: ui-packages/packages/management-console/server/MockData/graphql.js [hunk 7]
```diff
@@ -979,6 +1047,28 @@ module.exports = data = [
...
+      },
+      {
+        id: '27107f38-d888-4edf-9a4f-11b9e6d75m66',
+        name: 'Milestone 2: Order shipped',
+        status: 'AVAILABLE'
+      },
+      {
+        id: '27107f38-d888-4edf-9a4f-11b9e6d75m88',
+        name: 'Milestone 3: Order delivered',
+        status: 'TERMINATED'
```
Comment (nmirasch) [inline ui-packages/packages/management-console/server/MockData/graphql.js]: @Sara4994 we need to remove this mock milestone with an impossible milestone status

File: ui-packages/packages/management-console/server/MockData/types.js [hunk 8]
```diff
@@ -110,6 +111,18 @@ module.exports = typeDefs = gql`
...
+  type Milestones {
+    id: String!
+    name: String!
+    status: MilestoneStatus!
+  }
+  enum MilestoneStatus {
+    ACTIVE
+    AVAILABLE
+    COMPLETED
+    TERMINATED
```
Comment (nmirasch) [inline ui-packages/packages/management-console/server/MockData/types.js]: @Sara4994 here the same

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/ProcessDetailsMilestones.tsx (hunk 3 above)
Comment (Sara4994) [reply ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/ProcessDetailsMilestones.tsx:28]: @nmirasch ok.. i will remove Terminated

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/tests/__snapshots__/ProcessDetailsMilestones.test.tsx.snap (hunk 2 above)
Comment (Sara4994) [reply ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/tests/__snapshots__/ProcessDetailsMilestones.test.tsx.snap]: fixed

File: ui-packages/packages/management-console/server/MockData/graphql.js (hunk 7 above)
Comment (Sara4994) [reply ui-packages/packages/management-console/server/MockData/graphql.js]: removed

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/ProcessDetailsMilestones.tsx (hunk 4 above)
Comment (Sara4994) [reply ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/ProcessDetailsMilestones.tsx:47]: order: Active, Available, Completed

**Review by AjayJagan — APPROVED** [review]
LGTM

File: ui-packages/packages/management-console/src/components/Templates/ProcessDetailsPage/tests/__snapshots__/ProcessDetailsPage.test.tsx.snap [hunk 9]
```diff
@@ -679,6 +701,35 @@ exports[`Process Details Page component tests snapshot testing in Active state 1
...
+                          },
+                          Object {
+                            "id": "27107f38-d888-4edf-9a4f-11b9e6d75i86",
+                            "name": "Manager decision",
+                            "status": "COMPLETED",
+                          },
+                        ]
+                      }
+                    >
+
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Templates/ProcessDetailsPage/tests/__snapshots__/ProcessDetailsPage.test.tsx.snap]: why is the ProcessDetailsMilestones rendered with whitespace contents?

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/__mocks__/ProcessDetailsMilestones.tsx [hunk 10]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+const ProcessDetailsMilestones = () => {
+  return <> </>;
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsMilestones/__mocks__/ProcessDetailsMilestones.tsx]: I don't think we need the space in the fragment (results in whitespace in snapshots where this mock is used).

File: ui-packages/packages/management-console/src/components/Templates/ProcessDetailsPage/tests/__snapshots__/ProcessDetailsPage.test.tsx.snap (hunk 9 above)
Comment (jstastny-cz) [reply ui-packages/packages/management-console/src/components/Templates/ProcessDetailsPage/tests/__snapshots__/ProcessDetailsPage.test.tsx.snap]: caused by whitespace in the mock.

Comment (Sara4994) [reply ui-packages/packages/management-console/src/components/Templates/ProcessDetailsPage/tests/__snapshots__/ProcessDetailsPage.test.tsx.snap]: @jstastny-cz oopss.. thats my bad.. removed it now

**Review by jstastny-cz — APPROVED** [review]
Thanks.

**Review by nmirasch — APPROVED** [review]

**Comment by Sara4994** [comment]
jenkins please retest this

**Review by nmirasch — APPROVED** [review]

**Pull request merged**